src/test/librbd/fsx.cc

```diff
@@ -1794,6 +1834,82 @@ do_punch_hole(unsigned offset, unsigned length)
 	memset(good_buf + max_offset, '\0', max_len);
 }
 
+void
+dowritesame(unsigned offset, unsigned size, unsigned data_size)
+{
+	ssize_t ret;
+	off_t newsize;
+	unsigned buf_off;
+	int n;
+
+	offset -= offset % writebdy;
+	if (o_direct)
+		size -= size % writebdy;
+	if (size == 0 || data_size == 0) {
+		if (!quiet && testcalls > simulatedopcount && !o_direct)
+			prt("skipping zero size writesame\n");
+		log4(OP_SKIPPED, OP_WRITESAME, offset, size);
+		return;
+	}
+
+	if (size % data_size != 0) {
```

Comment:
If you generated the data_size here (tweaking the size if needed) you could eliminate this condition.